src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -61,6 +65,53 @@ private boolean isFileContentEqual(
     }
   }
 
+  @Test
+  public void testClientInfoConnectionProperty() throws Throwable {
+    Connection connection = null;
+    Statement statement = null;
+    ResultSet res = null;
+
+    try {
+      Properties props = new Properties();
+      props.put(
+          "snowflakeClientInfo",
+          "{\"spark.version\":\"3.0.0\", \"spark.snowflakedb.version\":\"2.8.5\", \"spark.app.name\":\"SnowflakeSourceSuite\", \"scala.version\":\"2.12.11\", \"java.version\":\"1.8.0_221\", \"snowflakedb.jdbc.version\":\"3.13.2\"}");
+      connection = getConnection(DONT_INJECT_SOCKET_TIMEOUT, props, false, false);
+      statement = connection.createStatement();
+      res = statement.executeQuery("select current_session_client_info()");
+      assertTrue(res.next());
+      String clientInfoJSONStr = res.getString(1);
+      JsonNode clientInfoJSON = mapper.readTree(clientInfoJSONStr);
+      // assert that spart version and spark app are found
+      assertEquals("spark version mismatch", "3.0.0", clientInfoJSON.get("spark.version").asText());
+      assertEquals(
+          "spark app mismatch",
+          "SnowflakeSourceSuite",
+          clientInfoJSON.get("spark.app.name").asText());
```

Comment:
Please also check below. This is property name GS determines whether the session is from Spark Connector.
```
      assertEquals(
          "snowflakedb version mismatch",
          "2.8.5",
          clientInfoJSON.get("spark.snowflakedb.version").asText());
```